### PR TITLE
docs: update container image dependencies and distribution process

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

### CODE REVIEW

File: Documentation/contributing/development/images.rst [hunk 1]
```diff
@@ -32,19 +32,70 @@ You can then push the image tag to your own registry for development builds:
...
+|github.com/cilium/cilium      | quay.io/cilium/cilium-runtime            |   Y      N     |    Quay auto     |
+|                              | quay.io/cilium/cilium-builder            |   Y      N     |    Quay auto     |
+|                              | [docker|quay].io/cilium/cilium           |   Y      N     | Quay/Docker auto |
+|                              | [docker|quay].io/cilium/docker-plugin    |   Y      N     | Quay/Docker auto |
+|                              | [docker|quay].io/cilium/hubble-relay     |   Y      N     | Quay/Docker auto |
+|                              | [docker|quay].io/cilium/operator         |   Y      N     | Quay/Docker auto |
+|                              | [docker|quay].io/cilium/operator-aws     |   Y      N     | Quay/Docker auto |
+|                              | [docker|quay].io/cilium/operator-azure   |   Y      N     | Quay/Docker auto |
+|                              | [docker|quay].io/cilium/operator-generic |   Y      N     | Quay/Docker auto |
++------------------------------+------------------------------------------+----------------+------------------+
```
Comment (pchaigno) [inline Documentation/contributing/development/images.rst]: That table is weirdly rendered, none of the cells are on several lines, which makes it hard to read.

**Review by pchaigno — CHANGES_REQUESTED** [review]
One rendering issue below. The doc build also found a couple spelling issues. LGTM otherwise!

File: Documentation/contributing/development/images.rst [hunk 2]
```diff
@@ -32,19 +32,84 @@ You can then push the image tag to your own registry for development builds:
...
+The following table contains the main container image repositories managed by
+Cilium team. It is planned to convert the build process for all images based
+on GH actions.
+
++-------------------------------+---------------------------------------------+------------------------------------------+-------------------------+-------------------+
+|     **Github Repository**     |                **Dockerfile**               |      **container image repository**      |   **Architectures**     | **Build process** |
+|                               |                                             |                                          +-----------+-------------+                   |
+|                               |                                             |                                          | **amd64** | **aarch64** |                   |
++-------------------------------+---------------------------------------------+------------------------------------------+-----------+-------------+-------------------+
+| github.com/cilium/cilium      | contrib/packaging/docker/Dockerfile.runtime | quay.io/cilium/cilium-runtime            |     Y     |      N      |     Quay auto     |
```
Comment (pchaigno) [inline Documentation/contributing/development/images.rst:53]: Nit: Might be worth removing `github.com` to gain a little space here. `contrib/packaging/docker` is also making the second column longer than necessary. It's not going to make the table fit the page though, so probably not worth re-pushing.

**Review by pchaigno — APPROVED** [review]
The table with all information is awesome! (It's also great to highlight that our Docker image build setup is a bit of a mess at the moment :smile: )

File: Documentation/contributing/development/images.rst [hunk 3]
```diff
@@ -32,19 +32,84 @@ You can then push the image tag to your own registry for development builds:
...
+|                               +---------------------------------------------+------------------------------------------+-----------+-------------+-------------------+
+|                               | images/startup-script/Dockerfile            | docker.io/cilium/startup-script          |     Y     |      Y      |     GH Action     |
++-------------------------------+---------------------------------------------+------------------------------------------+-----------+-------------+-------------------+
+
+Image dependency:
 
 ::
 
-    docker push cilium/cilium:v1.4.0
+    [docker|quay].io/cilium/cilium
```
Comment (joestringer) [inline Documentation/contributing/development/images.rst:92]: huh, useful. Is this generated with some CLI?

File: Documentation/spelling_wordlist.txt [hunk 4]
```diff
@@ -2,6 +2,7 @@ Admin
 Alemayhu
 Ansible
 Apiserver
+Archs
```
Comment (joestringer) [inline Documentation/spelling_wordlist.txt:5]: Where is `Archs` used?

**Review by joestringer — APPROVED** [review]
Looks good, awesome to get this documented.

Minor nit, I feel like the order of "git repo | container repo | dockerfile | architecture | build process" would make more sense but this also doesn't make any real difference so I'm fine with it as-is.

**Pull request merged**